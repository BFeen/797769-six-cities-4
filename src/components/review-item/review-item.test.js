import React from "react";
import renderer from "react-test-renderer";
import ReviewItem from "./review-item.jsx";


const mockReview = {
  offerId: 1,
  userName: `John`,
  userAvatar: `https://api.adorable.io/avatars/128/5`,
  rating: 5,
  description: `Best place of the world!`,
  dateTime: `December 2012`
};

describe(`ReviewItem snapshot testing`, () => {
  it(`Review item rendering correctly`, () => {
    const tree = renderer.create(
        <ReviewItem
          review={mockReview}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
