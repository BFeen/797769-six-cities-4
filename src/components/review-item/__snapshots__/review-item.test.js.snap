// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem snapshot testing Review item rendering correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="John"
        className="reviews__avatar user__avatar"
        height="54"
        src="img"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      John
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      Best place of the world!
    </p>
    <time
      className="reviews__time"
      dateTime="2019-04-24"
    >
      December 2012
    </time>
  </div>
</li>
`;
